src/wp-includes/interactivity-api/interactivity-api.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Interactivity API: Functions and hooks
+ *
+ * @package WordPress
+ * @subpackage Interactivity API
+ */
+
+/**
+ * Processes the directives on the rendered HTML of the interactive blocks.
+ *
+ * This processes only one root interactive block at a time because the
+ * rendered HTML of that block contains the rendered HTML of all its inner
+ * blocks, including any interactive block. It does so by ignoring all the
+ * interactive inner blocks until the root interactive block is processed.
+ *
+ * @since 6.5.0
+ *
+ * @param array $parsed_block The parsed block.
+ * @return array The same parsed block.
+ */
+function wp_interactivity_process_directives_of_interactive_blocks( $parsed_block ) {
+	static $root_interactive_block = null;
```

Comment:
Will this necessarily be the root of all interactive blocks? What if there are two interactive blocks that are siblings? Won't only the first then get processed?

Reply:
It works fine because the `render_block` method of the first root block is always called before the `render_block_data` of the next sibling one.

There are some tests that cover that case, but if you can think of other edge cases, please let me know.

https://github.com/WordPress/wordpress-develop/blob/69c6f78bd2a26246d0980497931042b43f8cad85/tests/phpunit/tests/interactivity-api/interactivity-api.php#L89-L106